app/src/sharedTest/java/org/oppia/app/testing/NavigationDrawerTestActivityTest.kt

```diff
@@ -0,0 +1,140 @@
+package org.oppia.app.testing
+
+import android.widget.TextView
+import androidx.test.core.app.ActivityScenario
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.contrib.DrawerActions.close
+import androidx.test.espresso.contrib.DrawerActions.open
+import androidx.test.espresso.contrib.DrawerMatchers.isClosed
+import androidx.test.espresso.contrib.DrawerMatchers.isOpen
+import androidx.test.espresso.matcher.ViewMatchers.isCompletelyDisplayed
+import androidx.test.espresso.matcher.ViewMatchers.withContentDescription
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.espresso.matcher.ViewMatchers.withParent
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import org.hamcrest.Matchers.allOf
+import org.hamcrest.Matchers.instanceOf
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.app.R
+import org.oppia.app.recyclerview.RecyclerViewMatcher
+
+/** Tests for [NavigationDrawerTestActivity]. */
+@RunWith(AndroidJUnit4::class)
+class NavigationDrawerTestActivityTest {
```

Comment:
Although the prod activities don’t support landscape currently, can you also add config change tests to make sure the desired behavior happens? Eg that the drawer works correctly in landscape for at least one test scenario, and that rotating closes the drawer?

Reply:
@BenHenning Added test cases for landscape mode test.
Opened drawer stays open on orientation change, rotating dose not closes the drawer ... do you want me to change this?

Reply:
Hmm. Let’s keep that behavior. Despite my earlier comment, I don’t have a compelling reason for closing the drawer on config changes. I can see either scenario as being okay.